nautobot/core/api/views.py

```diff
@@ -859,6 +861,49 @@ def get(self, request):
         return Response(object_counts)
 
 
+class GetSettingsView(NautobotAPIVersionMixin, APIView):
+    """
+    This view exposes Nautobot settings.
+
+    By default, it returns all serializable settings available in the system.
+    You can filter the settings by providing one or more `name` query parameters.
+
+    Example:
+
+    - /api/settings/            # Returns all serializable settings
+    - /api/settings/?name=FOO   # Returns the serializable setting with name 'FOO'
+    """
+
+    permission_classes = [IsAuthenticated]
+
+    def is_serializable(self, value):
+        with contextlib.suppress(TypeError):
+            json.dumps(value)
+            return True
+        return False
+
+    @extend_schema(exclude=True)
+    def get(self, request):
+        filter_params = request.GET.getlist("name")
+
+        # As of now, we just have serializable settings.  Because the purpose of this API is limited for the time being,
+        # we may need more access to serializable and non-serializable settings data as the new UI expands. As a result,
+        # serializing currently non-serializable data would be desirable in the future.
```

Comment:
Would serializing currently non-serializable data require a new parser? Or do we already have a parser that can do that?

Reply:
I dont think we have anything that currently do this